### PR TITLE
add support for "as sysdba/sysoper" connections

### PR DESCRIPTION
**Pull request by pvelder**
Here are the changes to allow "as sysdba/sysoper" kind of connections: 

    sys/syspwd@host:1521/ORCL?loc=America%2FLos_Angeles as sysdba

### CODE REVIEW

File: oci8.go [hunk 1]
```diff
@@ -500,6 +545,7 @@ func ParseDSN(dsnString string) (dsn *DSN, err error) {
 
 		}
 	}
+	// fmt.Printf("dsn: %v \n", dsn)
```
Comment (mattn) [inline oci8.go]: remove this line

File: oci8.go [hunk 2]
```diff
@@ -435,6 +475,10 @@ func ParseDSN(dsnString string) (dsn *DSN, err error) {
 		dsnString = dsnString[len(prefix):]
 	}
 
+	dsnString, mode := splitRight(dsnString, " as ")
+	// dsn.mode = toAuthMode(mode)
```
Comment (mattn) [inline oci8.go]: remove this commented line

File: oci8.go [hunk 3]
```diff
@@ -435,6 +475,10 @@ func ParseDSN(dsnString string) (dsn *DSN, err error) {
 		dsnString = dsnString[len(prefix):]
 	}
 
+	dsnString, mode := splitRight(dsnString, " as ")
```
Comment (mattn) [inline oci8.go]: please use query line `as=XXX` instead of ` as ` suffix. see oci8.go:453

Comment (pvelder) [reply oci8.go]: OK, changes are now available. 

I've also added a new as_sysdba example, using the as=sysdba query line with a comment line to compare with the "sqlplus as sysdba" usage:  "sqlplus sys/syspwd@tnsentry as sysdba"

**Comment by mattn** [comment]
Thank you. I'll check it on Linux and Windows in later.

**Comment by pvelder** [comment]
Ok thanks.  (Meanwhile I've already done some more tests on OSX and Windows. No problems yet..)

Kind regards,
Peter

**Comment by mattn** [comment]
Sorry about my delay. I'll look this in few days.

**Comment by mattn** [comment]
Thank you.

**Comment by pvelder** [comment]
No problem. Thanks for the merge :)

**Pull request merged**